autopush_rs/src/client.rs

```diff
@@ -570,15 +615,18 @@ where
         rotate_message_table: bool,
         check_storage: bool,
         connected_at: u64,
+        services: &Vec<Service>,
```

Comment:
FWIW `&[T]` tends to be more idiomatic rather than `&Vec<T>`

Reply:
Ah, is that used only when its a reference, or just in general as its shorter than Vec<T>?

Reply:
He just meant for &Vec<T>. If you're not doing anything too Vec specific, more types are able to deref into &[T] so it's good practice to prefer the slice type. I just linked this doc so, here's its further explanation https://deterministic.space/elegant-apis-in-rust.html#use-conversion-traits